README.md

```diff
@@ -476,3 +509,55 @@ Finally, digital signature helps us use Content Delivery Network without
 worrying that a CDN compromise would end-up serving bad data to our users.
 Signing at the source reduces the pressure off of the infrastructure and
 allows us to rely on vendors without worrying about data integrity.
+
+### Algorithm
+
+## Addon Signing (XPI)
+
+### Algorithm
+
+Addons are XPI files (zip archives) where each file in the archives is hashed
+with SHA-1 and MD5, and the manifest containing all the hashes is signed. To
+request a signature, the client must first compute the manifest file (stored in
+`META-INF/manifest.mf` in the XPI), then make a signature file (stored in
+`META-INF/mozilla.sf`) and submit the `sf` file to the signing API, alongside
+the addon ID (typically an email address). The signing API returns an S/MIME
```

Comment:
The addon ID looks like an email address but it rarely is a real email. Developers tend to use `any-kind-of-id` at their official add-on domain. Example: `public-release@ublock.org`.